File: P226 Invert Binary Tree/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public TreeNode invertTree(TreeNode root) {
        if (root == null) {
            return null;
        }
        TreeNode right = invertTree(root.right);
        TreeNode left = invertTree(root.left);
        root.left = right;
        root.right = left;
        return root;
    }
    /**
     * Iterative Version
     * 
     * if (root == null) return null;
     * Queue<TreeNode> queue = new LinkedList<TreeNode>();
     * queue.add(root);
     * while (!queue.isEmpty()) {
     *    TreeNode current = queue.poll();
     *    TreeNode temp = current.left;
     *    current.left = current.right;
     *    current.right = temp;
     *    if (current.left != null) queue.add(current.left);
     *    if (current.right != null) queue.add(current.right);
     * }
     * return root;
     */
}

File: P226 Invert Binary Tree/Solution.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    # @param {TreeNode} root
    # @return {TreeNode}
    def invertTree(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        if root is None:
            return None
        if root != None:
            temp = root.left
            root.left = root.right
            root.right = temp
            self.invertTree(root.left)
            self.invertTree(root.right)
        return root
    ''' Iterative version 
    # @param {TreeNode} root
    # @return {TreeNode}
    def invertTree(self, root):
        if root is None:
            return None
        queue = [root]
        while queue:
            front = queue.pop(0)
            if front.left:
                queue.append(front.left)
            if front.right:
                queue.append(front.right)
            front.left, front.right = front.right, front.left
        return root
    '''